test/option_parser.coffee

```diff
@@ -65,3 +71,37 @@ test "throw if multiple flags try to use the same short or long name", ->
     ['--just-long',       'desc']
     ['-j', '--just-long', 'another desc']
   ]
+
+reQuote = (str) -> (str ? '').replace /[.?*+^$[\]\\(){}|-]/g, "\\$&"
+
+flagPattern = (flag) ->
+  switch flag.length
+    when 2
+      shortFlag = null
+      [longFlag, desc] = flag
+    when 3
+      [shortFlag, longFlag, desc] = flag
+    else
+      throw new Error "invalid flag"
+  longFlagPat = reQuote longFlag.match(/--[^\s]+/)[0]
+  descPat = reQuote desc
+  if shortFlag?
+    joined = [reQuote(shortFlag), longFlagPat, descPat].join '.*'
+  else
+    joined = [longFlagPat, descPat].join '.*'
+  new RegExp "^.*#{joined}.*$", 'm'
+
+test "help text shows banner and all switches", ->
+  bannerHelp = opt.help()
+  bannerPattern = new RegExp reQuote(banner), 'g'
+  ok bannerHelp.match bannerPattern
+  for flag in flags
+    linePat = flagPattern flag
+    ok bannerHelp.match linePat
```

Comment:
This tests that the option parser correctly outputs a banner text string, but not necessarily that `coffee -h` actually outputs the correct help text. For that, you need to actually test `opt.help()` against the help text, _which should be written here in the test._ I know it doesn’t feel DRY to repeat a string that is elsewhere in the codebase, but if that string changes in the main code, we _want_ this test to fail—either to inform us that we unexpectedly changed the help text, or to remind us to update the test, too, so that we preserve this check for the future. The same applies to the list of switches.